File: ItineraryBuilder-Backend/Models/ItineraryPlace.cs
using System.Text.Json.Serialization;

namespace ItineraryBuilder_Backend.Models
{
    public class ItineraryPlace
    {
        public int Id { get; set; } 
        public int Number { get; set; }
        public int VisitTime { get; set; }
        public int IsVisited { get; set; }  
        public int PlaceId { get; set; } // Foreign key to Place

        public Place Place { get; set; } // Navigation property
        public int ItineraryId { get; set; } // Foreign key to Itinerary
        [JsonIgnore]
        public Itinerary Itinerary { get; set; } // Navigation property
    }

}

File: itinerarybuilder/lib/models/itinerary_place.dart
import 'package:itinerarybuilder/models/place.dart';
import 'package:json_annotation/json_annotation.dart';
part 'itinerary_place.g.dart';
@JsonSerializable()
class ItineraryPlace {
  String Id;
  int Number;
  int VisitTime;
  int IsVisited;
  String PlaceId;
  String ItineraryId;

  ItineraryPlace({
    required this.Id,
    required this.Number,
    required this.VisitTime,
    required this.IsVisited,
    required this.PlaceId,
    required this.ItineraryId
  });

  factory ItineraryPlace.fromJson(Map<String, dynamic> json) => _$ItineraryPlaceFromJson(json);
  Map<String, dynamic> toJson() => _$ItineraryPlaceToJson(this);
}